go-controller/pkg/controller/controller.go

```diff
@@ -214,3 +237,97 @@ func (c *controller[T]) ReconcileAll() {
 		c.queue.Add(key)
 	}
 }
+
+// SubController is a controller that should be used as a part of MultiController.
+// It is very similar to Controller, but it doesn't have its own initialSync method.
+// Initial sync is handled by MultiController instead.
+// SubController is not meant to be used standalone, all its methods are private and will be called by MultiController.
+type SubController interface {
+	addHandler() error
+	startWorkers(threadiness int) error
+	stop()
+}
+
+func NewSubController[T any](name string, config *Config[T]) SubController {
+	return &controller[T]{
+		name:   name,
+		config: config,
+		queue: workqueue.NewRateLimitingQueueWithConfig(
+			config.RateLimiter,
+			workqueue.RateLimitingQueueConfig{
+				Name: name,
+			},
+		),
+		stopChan: make(chan struct{}),
+		wg:       &sync.WaitGroup{},
+	}
+}
+
+// MultiController manages multiple controllers.
+// It is only needed when initial sync is required across multiple informers.
+// If your intialSync method `List`s more than one object type, it is very likely you should use MultiController.
+// InitialSync will be run after all subcontroller event handlers are added to ensure proper event handling
+// for related informers.
+type MultiController interface {
+	Start(threadiness int) error
+	Stop()
+}
```

Comment:
So this is not compliant with the Controller interface. No way to ReconcileAll or anything?